nx/lib/nx/binary_backend.ex

```diff
@@ -1038,6 +1038,152 @@ defmodule Nx.BinaryBackend do
     from_binary(out, output_data)
   end
 
+  @impl true
+  def qr(
+        {%{shape: {m, k} = shape_q, type: output_type} = q_holder,
+         %{shape: {k, n} = shape_r, type: output_type} = r_holder},
+        %{type: input_type} = tensor,
+        opts
+      ) do
+    mode = opts[:mode]
+
+    r =
+      if mode == :reduced do
+        tensor[[0..(k - 1), 0..(n - 1)]]
```

Comment:
You can slice the binary in the same way as `Nx.slice` instead of having to dispatch outside here, but it's not that much overhead

Reply:
It turns out that:
a) This particular slicing was much more efficient on the binary because it happens to just be a "take N bits from the beggining of the binary" algorithm
b) Only this first slicing is needed. The other one at the end of the function was redundant in the end.